### PR TITLE
executor: migrate test-infra to testify for inspection_result_test.go

### LINKED ISSUE #28590: migrate test-infra to testify for `inspection_result_test.go`

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #28590 

```release-note
None
```

### CODE REVIEW

**Comment by tisonkun** [comment]
test failed exactly on changes:

```
[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-464)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-465)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-466)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-467)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-468)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-469)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-470)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-471)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-472)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-473)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-474)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-475)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-476)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-477)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-478)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-479)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-480)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-481)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-482)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-483)[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36731/pipeline#step-59-log-484)[2022-02-21T10:08:18.610Z] --- FAIL: TestCriticalErrorInspection (0.69s)

[2022-02-21T10:08:18.610Z]     inspection_result_test.go:613: 

[2022-02-21T10:08:18.610Z]         	Error Trace:	inspection_result_test.go:613

[2022-02-21T10:08:18.610Z]         	Error:      	Not equal: 

[2022-02-21T10:08:18.610Z]         	            	expected: 0x0

[2022-02-21T10:08:18.610Z]         	            	actual  : 0x2

[2022-02-21T10:08:18.610Z]         	Test:       	TestCriticalErrorInspection

[2022-02-21T10:08:18.610Z]         	Messages:   	unexpected warnings: [{Level:Warning Err:tikv node store1 does not contain status address

[2022-02-21T10:08:18.610Z]         	            	github.com/pingcap/tidb/executor.(*clusterLogRetriever).startRetrieving

[2022-02-21T10:08:18.610Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/memtable_reader.go:574

[2022-02-21T10:08:18.610Z]         	            	github.com/pingcap/tidb/executor.(*clusterLogRetriever).initialize

[2022-02-21T10:08:18.610Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/memtable_reader.go:545

[2022-02-21T10:08:18.610Z]         	            	github.com/pingcap/tidb/executor.(*clusterLogRetriever).retrieve

[2022-02-21T10:08:18.610Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/memtable_reader.go:637

[2022-02-21T10:08:18.610Z]         	            	github.com/pingcap/tidb/executor.(*MemTableReaderExec).Next

[2022-02-21T10:08:18.610Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/memtable_reader.go:118

[2022-02-21T10:08:18.610Z]         	            	github.com/pingcap/tidb/executor.Next

[2022-02-21T10:08:18.610Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/executor.go:301

[2022-02-21T10:08:18.610Z]         	            	github.com/pingcap/tidb/executor.(*projectionInputFetcher).run

[2022-02-21T10:08:18.610Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/projection.go:381

[2022-02-21T10:08:18.610Z]         	            	runtime.goexit

[2022-02-21T10:08:18.610Z]         	            		/usr/local/go/src/runtime/asm_amd64.s:1371} {Level:Warning Err:rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp :10080: connect: connection refused"}]
```

File: executor/inspection_result_test.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ import (
 func TestInspectionResult(t *testing.T) {
 	store, clean := testkit.CreateMockStore(t)
 	defer clean()
-	tk := testkit.NewTestKit(t, store)
+	tk := testkit.NewTestKitWithInit(t, store)
```
Comment (tisonkun) [inline executor/inspection_result_test.go:42]: Generally we just write:

```go
tk := testkit.NewTestKit(t, store)
tk.MustExec("use test")
```

**Comment by tisonkun** [comment]
I'm going to push a patch onto this PR...

**Review by tisonkun — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**